### PR TITLE
Use `impl Into<Stroke>` as argument in a few more places

### PR DESCRIPTION
**Pull request by Phen-Ro**
* Functions that take Stroke were updated to take Into<Stroke> to make them consistent with other Into<Stroke> parameters.
* Vec2 implements DivAssign<f32>, to make it consistent with already implementing MulAssign<f32> and Div<f32>.
* Vec2::angled() uses sin_cos() rather than an individual sin() and cos() call for an immeasurable but hypothetical performance improvement.
* Disable the lock_reentry_single_thread() mutex test.  Lock()ing twice on the same thread is not guaranteed to panic.

* Closes <https://github.com/emilk/egui/issues/3419>.

### CODE REVIEW

**Comment by Phen-Ro** [comment]
I was unable to complete scripts/check.sh because the winit crate failed with "The platform you're compiling for is not supported by winit".  I hope you don't mind that I'll move this PR to ready anyway.

**Comment by Phen-Ro** [comment]
In particular, these were the lines that failed complaining about winit platform support:
`(cd crates/eframe && cargo check --no-default-features --features "wgpu")`
`(cd crates/egui_demo_app && cargo check --no-default-features --features "wgpu")`
`(cd crates/egui_glow && cargo check --all-features)`

Everything else passed.

File: crates/epaint/src/mutex.rs [hunk 1]
```diff
@@ -389,6 +389,7 @@ mod tests {
 
     #[test]
     #[should_panic]
+    #[ignore = "Documentation does not gaurantee panic. \"The exact behavior on locking a mutex in the thread which already holds the lock is left unspecified. However, this function will not return on the second call (it might panic or deadlock, for example).\""]
```
Comment (emilk) [inline crates/epaint/src/mutex.rs:392]: Interesting! Then let's remove the test altogether instead

**Review by emilk — APPROVED** [review]

**Comment by Phen-Ro** [comment]
Merge conflict resolved and mutex test removed.

**Pull request merged**